.github/workflows/checks.yml

```diff
@@ -0,0 +1,39 @@
+name: Checks
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  gradle:
+    strategy:
+      matrix:
+#        restore these once cmake is fixed
+#        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        os: [ macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Set up Python 3.7.12
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.12
+
+      - name: Install devnet
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Run tests
+        run: ./gradlew :lib:test --stacktrace --info
```

Comment:
I wonder if we need `--stacktrace` and `--info` in the actual setup. I've added this as part of the debugging, but maybe it makes sense to keep them?